### PR TITLE
Issue #13501 - printEvent Source to much more cleaner

### LINKED ISSUE #13501: printEvent Source to much more cleaner
Greetings,

currently, when doing `kubectl describe node node-name` then in `From` column, the ui result is 

```
{kubelet gke-cluster-1-micro-a0f27b19-node-vn5m}
```

maybe can do something like this instead? remove `{}` then join it with `,`

```
kubelet, gke-cluster-1-micro-a0f27b19-node-vn5m
```

### PR DESCRIPTION
**Pull request by matthyx**
Create a formatEventSource func and use it.

**What this PR does / why we need it**:
Adds a smarter way to print EventSource, removing "{ }" and the eventual extra ", " when Host is empty.
It was also reported in OpenShift issue https://github.com/openshift/origin/issues/6586

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #13501 

**Special notes for your reviewer**:
I have used this new func in all user facing printouts, but there are other places where EventSource are printed to logs, like DumpEventsInNamespace in test/e2e/framework/util.go for instance.
Don't know if I should correct there too...

**Release note**:

```release-note
Improve formatting of EventSource in kubectl get and kubectl describe
```

### CODE REVIEW

**Comment by matthyx** [comment]
I signed it!

**Comment by matthyx** [comment]
I signed it!
Looks like my first commit used a bogus email from my machine hostname, should be fixed now.

**Comment by janetkuo** [comment]
@k8s-bot ok to test

File: pkg/api/helpers.go [hunk 1]
```diff
@@ -615,3 +615,10 @@ func PodAnnotationsFromSysctls(sysctls []Sysctl) string {
 	}
 	return strings.Join(kvs, ",")
 }
+
+// FormatEventSource formats EventSource as a comma separated string excluding Host when empty
+func FormatEventSource(es EventSource) string {
+	EventSourceString := []string{es.Component}
+	if len(es.Host) > 0 { EventSourceString = append(EventSourceString, es.Host) }
```
Comment (janetkuo) [inline pkg/api/helpers.go]: `gofmt` this

Comment (janetkuo) [inline pkg/api/helpers.go]: Suggest moving this function to resource_printer.go and not exposing it

Comment (matthyx) [inline pkg/api/helpers.go]: I have moved the function to resource_printer.go
How do I not expose it?

**Comment by matthyx** [comment]
@k8s-bot cvm gke e2e test this

**Comment by matthyx** [comment]
@k8s-bot cvm gke e2e test this

**Comment by matthyx** [comment]
@k8s-bot gci gke e2e test this

File: pkg/api/helpers.go (hunk 1 above)
Comment (janetkuo) [inline pkg/api/helpers.go]: I meant lower-casing it

**Review by janetkuo — APPROVED** [review]

**Comment by janetkuo** [comment]
@matthyx LGTM, thanks much! Would you squash commits?

**Comment by matthyx** [comment]
@janetkuo I have squashed commits. Thanks for your review :)

**Comment by matthyx** [comment]
@k8s-bot bazel test this

**Comment by janetkuo** [comment]
Submit Queue complained 'The PR was changed after the approved label was added.' Re-applying label 

/approved

**Pull request merged**